File: clicker/app/src/main/java/com/example/myapplication/MainActivity2.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity2 extends AppCompatActivity {

    private Button back;
    private TextView text;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        text = findViewById(R.id.textview1);
        back = findViewById(R.id.Button);
        Intent intent = getIntent();

        int a2 = intent.getExtras().getInt("a");
        text.setText(String.valueOf(a2));

        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                finish();
            }
        });
    }
}

File: ConstraintLayout/app/src/main/java/com/example/constraintlayout/MainActivity2.kt
package com.example.constraintlayout

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class MainActivity2 : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        val intent = Intent(this,MainActivity3::class.java)
        startActivity(intent)
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        Toast.makeText(this,"흠",Toast.LENGTH_SHORT).show()
    }
}